app/dao/unsubscribe_request_dao.py

```diff
@@ -80,3 +105,20 @@ def create_unsubscribe_request_reports_dao(unsubscribe_request_report):
 def update_unsubscribe_request_report_processed_by_date_dao(report, report_has_been_processed):
     report.processed_by_service_at = datetime.utcnow() if report_has_been_processed else None
     db.session.add(report)
+
+
+@autocommit
+def assign_unbatched_unsubscribe_requests_to_report_dao(report_id, service_id, earliest_timestamp, latest_timestamp):
+    """
+    This method retrieves all the un-batched unsubscribe requests received before or on
+    the report's latest_timestamp and sets their unsubscribe_request_report_id to the report_i
+    """
+    unbatched_unsubscribe_requests = UnsubscribeRequest.query.filter(
+        UnsubscribeRequest.unsubscribe_request_report_id.is_(None),
+        UnsubscribeRequest.service_id == service_id,
+        UnsubscribeRequest.created_at >= earliest_timestamp,
+        UnsubscribeRequest.created_at <= latest_timestamp,
+    ).all()
```

Comment:
I don’t know if this is the exact SQLAlchemy syntax but a single update query should be much more efficient than a for loop
```suggestion
    ).update(
        report_id=report_id
    )
```